src/components/Channel/PlayerContainer.jsx

```diff
@@ -1,21 +1,39 @@
 import React from 'react';
 import { useRouter } from 'next/router';
 import styled from '@emotion/styled';
-import usePlayers from '../../hooks/channel/usePlayers';
 
+import { socketClient } from '../../hooks/socket/useSocket';
+import usePlayers from '../../hooks/channel/usePlayers';
 import PlayerItem from './PlayerItem';
 import ErrorBox from '../ErrorBox';
+import { EVENTS } from '../../constants/socketEvent';
 
 export default function PlayerContainer() {
   const {
     query: { channelId },
   } = useRouter();
-  const { players, error } = usePlayers(channelId);
+  const { players, error, mutate } = usePlayers(channelId);
+
+  if (channelId == null || players == null) {
+    return <></>;
+  }
 
   if (error) {
     return <ErrorBox message={error.message} />;
   }
 
+  socketClient.on(EVENTS.LISTEN_PLAYER_READY, ({ _id, userRole }) => {
+    const newPlayers = players.map((player) => {
+      if (player.userId._id === _id) {
+        player.characterId = userRole.characterId;
+      }
+
+      return player;
+    });
+
+    mutate((data) => ({ ...data, players: [...newPlayers] }));
```

Comment:
data 대신에 prevPlayers 라는 변수명이 적절할 것 같습니다.

newPlayers 타입이 배열이라, 굳이 spread해서 다시 배열에 담지 않아도 될 것 같습니다.
```suggestion
    mutate((prevPlayers) => ({ ...prevPlayers, players: newPlayers }));
```

Reply:
prevPlayers로 수정하겠습니다! 감사합니다🙇